samples/subsys/ipc/ipc_service/icmsg/src/main.c

```diff
@@ -8,7 +8,12 @@
 #include <zephyr/device.h>
 
 #include <zephyr/ipc/ipc_service.h>
+#if defined(CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP)
 #include <hal/nrf_reset.h>
+#elif defined(CONFIG_BOARD_STM32H747I_DISCO_M7)
+#include <stm32_ll_rcc.h>
+#include <stm32h7xx_ll_system.h>
```

Comment:
I might have missed an essential part here, but I don't see where these headers are required.

Reply:
You are absolutely right. There was a point at which there were some functions from there being used for testing and debugging, but I removed that by now. Will remove the headers then.